modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/ApplicationThrottlingResetTestCase.java

```diff
@@ -0,0 +1,294 @@
+                "3600", null, ApplicationKeyGenerateRequestDTO.KeyTypeEnum.PRODUCTION,
+                null, grantTypes);
+        String accessToken = applicationKeyDTO.getToken().getAccessToken();
+
+        Map<String, String> requestHeaders = new HashMap<String, String>();
+        requestHeaders.put("Authorization", "Bearer " + accessToken);
+        requestHeaders.put("accept", "text/xml");
+        requestHeaders.put("content-type", "application/json");
+        String apiInvocationUrl = getAPIInvocationURLHttps(APIContext + "/" + APIVersion + "/");
+
+        // invoke the api and check the throttling
+        checkThrottling(apiInvocationUrl, requestHeaders, 5);
+
+        String userId = requestCountApplicationDTO.getOwner();
+        if (userId != null && userId.contains("@")) {
+            userId = userId.substring(0, userId.lastIndexOf("@"));
+        }
+
+        // reset the application policy
+        org.wso2.am.integration.clients.store.api.ApiResponse<Void> resetResponse =
+                restAPIStore.resetApplicationThrottlePolicy(
+                requestCountApplicationDTO.getApplicationId(), userId);
+        Assert.assertEquals(resetResponse.getStatusCode(), 200, "reset Application policy is not successful");
+        Thread.sleep(5000);
+
+        // invoke the api and check the throttling again to verify reset is happened
+        checkThrottling(apiInvocationUrl, requestHeaders, 5);
+    }
+
+    @Test(groups = { "wso2.am" }, description = "Test reset application throttling policy with request count limit")
+    public void testResetPolicyWithBandwidthLimit() throws Exception {
+        //subscribe to API
+        SubscriptionDTO subscriptionDTO = restAPIStore.subscribeToAPI(apiId, bandwidthApplicationDTO.getApplicationId(),
+                Constants.TIERS_UNLIMITED);
+        Assert.assertEquals(subscriptionDTO.getThrottlingPolicy(), Constants.TIERS_UNLIMITED,
+                "Error occurred " + "while subscribing to the api. Subscribed policy is not as expected as "
+                + Constants.TIERS_UNLIMITED);
+
+        // generate keys and token
+        ArrayList<String> grantTypes = new ArrayList<>();
+        grantTypes.add(APIMIntegrationConstants.GRANT_TYPE.CLIENT_CREDENTIAL);
+        ApplicationKeyDTO applicationKeyDTO = restAPIStore.generateKeys(bandwidthApplicationDTO.getApplicationId(),
+                "3600", null, ApplicationKeyGenerateRequestDTO.KeyTypeEnum.PRODUCTION,
+                null, grantTypes);
+        String accessToken = applicationKeyDTO.getToken().getAccessToken();
+
+        Map<String, String> requestHeaders = new HashMap<String, String>();
+        requestHeaders.put("Authorization", "Bearer " + accessToken);
+        requestHeaders.put("accept", "text/xml");
+        requestHeaders.put("content-type", "application/json");
+        String apiInvocationUrl = getAPIInvocationURLHttps(APIContext + "/" + APIVersion + "/");
+
```

Comment:
Code untill here should go to setEnvironment method

Reply:
Fixed with https://github.com/wso2/product-apim/pull/13493/commits/ea0b01ff6b639e1245e49754bb332e3fcf359753